File: backend/src/main/java/com/eeum/dto/response/RoutineResponse.java
package com.eeum.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Instant;
import java.time.LocalTime;
import java.util.List;

@Getter
@AllArgsConstructor
public class RoutineResponse {
    private Integer routineId;
    private String name;
    private Boolean triggerType;
    private Integer routineWeekday;
    private String routineDescription;
    private LocalTime actTime;
    private Instant createdAt;
    private Instant updatedAt;
    private Integer iconId;
    private Boolean isAi;
    private String iconUrl;
    private List<RoutineDetailResponse> details;
}

File: frontend/app/src/main/java/com/example/eeum/data/model/response/routine/RoutineResponse.kt
package com.example.eeum.data.model.response.routine

import com.example.eeum.data.model.response.common.BaseResponse

data class RoutineResponse (
    val routineId: Int,
    val name: String,
    val triggerType: Boolean,
    val routineWeekday: Int,
    val routineDescription: String?,
    val actTime: String,
    val createdAt: String,
    val updatedAt: String,
    val details: List<BaseResponse>
)
